### ISSUE TITLE
"TypeError: rstrip arg must be None or str" when fetching from non-existing repo

### ISSUE THREAD

**Issue by inducer** [dulwich/dulwich#785]
When fetching from a non-existing git repo via  a Paramiko SSH git client, I get the following backtrace:

```
Traceback (most recent call last):
  File "/home/andreas/.cache/pypoetry/virtualenvs/relate-courseware-on7htJeR-py3.8/lib/python3.8/site-packages/dulwich/client.py", line 890, in fetch_pack
    refs, server_capabilities = read_pkt_refs(proto)
  File "/home/andreas/.cache/pypoetry/virtualenvs/relate-courseware-on7htJeR-py3.8/lib/python3.8/site-packages/dulwich/client.py", line 224, in read_pkt_refs
    for pkt in proto.read_pkt_seq():
  File "/home/andreas/.cache/pypoetry/virtualenvs/relate-courseware-on7htJeR-py3.8/lib/python3.8/site-packages/dulwich/protocol.py", line 266, in read_pkt_seq
    pkt = self.read_pkt_line()
  File "/home/andreas/.cache/pypoetry/virtualenvs/relate-courseware-on7htJeR-py3.8/lib/python3.8/site-packages/dulwich/protocol.py", line 212, in read_pkt_line
    raise HangupException()
dulwich.errors.HangupException: The remote server unexpectedly closed the connection.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/andreas/work/django/relate/course/versioning.py", line 548, in update_course
    run_course_update_command(
  File "/home/andreas/work/django/relate/course/versioning.py", line 347, in run_course_update_command
    fetch_pack_result = client.fetch(remote_path, repo)
  File "/home/andreas/.cache/pypoetry/virtualenvs/relate-courseware-on7htJeR-py3.8/lib/python3.8/site-packages/dulwich/client.py", line 430, in fetch
    result = self.fetch_pack(
  File "/home/andreas/.cache/pypoetry/virtualenvs/relate-courseware-on7htJeR-py3.8/lib/python3.8/site-packages/dulwich/client.py", line 892, in fetch_pack
    _remote_error_from_stderr(stderr)
  File "/home/andreas/.cache/pypoetry/virtualenvs/relate-courseware-on7htJeR-py3.8/lib/python3.8/site-packages/dulwich/client.py", line 754, in _remote_error_from_stderr
    lines = [line.rstrip(b'\n') for line in stderr.readlines()]
  File "/home/andreas/.cache/pypoetry/virtualenvs/relate-courseware-on7htJeR-py3.8/lib/python3.8/site-packages/dulwich/client.py", line 754, in <listcomp>
    lines = [line.rstrip(b'\n') for line in stderr.readlines()]
TypeError: rstrip arg must be None or str
```
It seems like something goes wrong in the error handling.

**Comment by inducer** [comment]
dulwich 0.20.3, I forgot to add. I would have added a CLI reproducer, however #784.

**Comment by jelmer** [comment]
It looks like the paramiko client is broken - it's providing a plain string stderr object rather than a bytes one.

**Comment by jelmer** [comment]
Actually, it looks like this is already fixed - can you try this with 0.20.5 ?

**Comment by jelmer** [comment]
On a related note, we should probably try to get the ParamikoSSHVendor into core - its current place in contrib means that it's essentially untested.
